qiskit/quantum_info/synthesis/qsd.py

```diff
@@ -253,3 +267,13 @@ def _apply_a2(circ):
     qc3 = two_qubit_decompose.two_qubit_cnot_decompose(mat2)
     ccirc.data[ind2] = (qc3.to_gate(), qargs, cargs)
     return ccirc
+
+
+def _is_block_diagonal(mat, dim_o2):
+    """
+    Check whether matrix is block diagonal
+
+    TODO: check for matrices which can be made block diagonal
+    """
+    zmat = np.zeros((dim_o2, dim_o2))
+    return np.allclose(mat[:dim_o2, dim_o2:], zmat) and np.allclose(mat[dim_o2:, :dim_o2], zmat)
```

Comment:
We can actually just use Numpy broadcasting to do `np.allclose(mat[:dim_o2, dim_o2:], 0)` etc, though it's not desperately important.

Reply:
Yeah, scalars could as array-like for the purposes of broadcasting.